src/map/ai/controllers/trust_controller.cpp

```diff
@@ -118,23 +142,61 @@ void CTrustController::DoCombatTick(time_point tick)
 
 void CTrustController::DoRoamTick(time_point tick)
 {
-    if (POwner->PMaster->PAI->IsEngaged() && GetTopEnmity() == POwner->PMaster)
+    auto PMaster = static_cast<CCharEntity*>(POwner->PMaster);
+    bool trustEngageCondition = (PMaster->GetBattleTarget() && PMaster->GetBattleTarget()->PLastAttacker == PMaster);
+
+    if (PMaster->PAI->IsEngaged() && trustEngageCondition)
     {
-        POwner->PAI->Internal_Engage(POwner->PMaster->GetBattleTargetID());
+        POwner->PAI->Internal_Engage(PMaster->GetBattleTargetID());
     }
 
-    float currentDistance = distance(POwner->loc.p, POwner->PMaster->loc.p);
+    uint8 currentPartyPos = GetPartyPosition();
+    CBattleEntity* PFollowTarget = (GetPartyPosition() > 0) ? (CBattleEntity*)PMaster->PTrusts.at(currentPartyPos - 1) : POwner->PMaster;
+    float currentDistance = distance(POwner->loc.p, PFollowTarget->loc.p);
+
+    for (auto POtherTrust : PMaster->PTrusts)
+    {
+        if (POtherTrust != POwner && distance(POtherTrust->loc.p, POwner->loc.p) < 1.0f)
```

Comment:
I'm wondering if it might be better to have declumping occur after distance checks so trusts aren't too preoccupied declumping from each other when they should pathing to the master.

(Or if kept here, add in a check here to make sure the _current_ trust isn't already following a path)

Reply:
I shall test and see how it looks 👍

Reply:
All the different combinations look about the same, so I changed to use your suggestions